ui/src/main/resources/edu/wpi/grip/ui/templates/Pipeline.vm

```diff
@@ -0,0 +1,100 @@
+#**
+This creates the complete java pipeline.
+*#
+#set($needsMut = false)
+#set($interpolation = false)
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.HashMap;
+import javax.imageio.ImageIO;
+import javax.swing.ImageIcon;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+
+import org.opencv.core.*;
+import org.opencv.core.Core.*;
+import org.opencv.imgcodecs.Imgcodecs;
+import org.opencv.imgproc.*;
+
+/**
+* Pipeline class.
+*
+* <P>Autogenerated pipeline from GRIP.
+*
+* <P>Make sure to set all sources using the setters before running processImage().
+*
+* <P>Tutorials and examples can be found online
+*
+* @author GRIP
+*/
+public class Pipeline{
+
+	//This map links the Outputs with their names
+	protected Map<String,Object> outputs;
+#foreach($source in $pipeline.getSources())
+    protected ${source.type()} $source.value();
+#end
+
+	/**
+	 * This constructor sets up the pipeline
+	 */
+	public Pipeline(){
+		outputs = new HashMap<String,Object>();
+	}
+
+	/**
+	 * This is the primary method that runs the entire pipeline and updates the outputs.
+	 */
+	public void processImage(){
+#set($count = 0)
+#foreach($step in $pipeline.getSteps())
+#set($c = "S"+$count)
+	    //Step$count: $step.name():
+#parse("src/main/resources/edu/wpi/grip/ui/templates/CV Step.vm")
+#set($count = $count + 1)
+
+#end
+}
+#foreach($source in $pipeline.getSources())
+      /**
+      * This method is a generated setter for $source.value().
+      * @param source is the $source.type() that the source will be set to
+      */
+      public void set${source.value()}(${source.type()} ${source.value()}){
+      	 this.$source.value() = ${source.value()};
+      }
+#end
+#foreach($step in $pipeline.getSteps())
+#foreach($output in $step.getOutputs())
+    	/**
+    	 * This method is a generated getter for the output of a $step.name().
+    	 * @return ${output.baseType()} is the output.
+    	 */
+      public ${output.baseType()} get${output.name()}(){
```

Comment:
The name should be more intuitive than `getoutputX`. Use the name of the operation (or at least it's position in the pipeline) and the name of the specific outputs -- e.g. `getStep1Dst`, or even better, `getHslThreshold1Dst` (if only one operation of that type is in the pipeline, the number should probably be dropped)